astropy/stats/funcs.py

```diff
@@ -728,26 +735,65 @@ def bootstrap(data, bootnum=100, samples=None, bootfunc=None):
         Bootstrapped data. Each row is a bootstrap resample of the data.
 
     """
+
     if samples is None:
         samples = data.shape[0]
 
     # make sure the input is sane
     assert samples > 0, "samples cannot be less than one"
     assert bootnum > 0, "bootnum cannot be less than one"
+    assert type(data) is np.ndarray, "data must be a numpy.ndarray"
```

Comment:
I believe using ``isinstance()`` is the recommended way to check whether the object type is the expected one.